x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshots.java

```diff
@@ -218,10 +200,15 @@ public void onIndexModule(IndexModule indexModule) {
 
     @Override
     public Optional<EngineFactory> getEngineFactory(IndexSettings indexSettings) {
-        if (SEARCHABLE_SNAPSHOTS_FEATURE_ENABLED
-            && isSearchableSnapshotStore(indexSettings.getSettings())
+        if (SearchableSnapshotsConstants.isSearchableSnapshotStore(indexSettings.getSettings())
             && indexSettings.getSettings().getAsBoolean("index.frozen", false) == false) {
-            return Optional.of(engineConfig -> new ReadOnlyEngine(engineConfig, null, new TranslogStats(), false, Function.identity()));
+            return Optional.of(engineConfig -> new ReadOnlyEngine(engineConfig, null, new TranslogStats(), false, Function.identity()) {
+                @Override
+                protected void ensureMaxSeqNoEqualsToGlobalCheckpoint(SeqNoStats seqNoStats) {
```

Comment:
Instead of overriding the method every time it's needed maybe we could pass a `ensureMaxSeqNoEqualsToGlobalCheckpoint` boolean flag when the ReadOnlyEngine is instanciated?

Reply:
Sure, I tried that in cf2577e225afb308767cd7a4b2605ca67395fb49.

Reply:
Thanks!